flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/SqlClientTest.java

```diff
@@ -156,6 +156,23 @@ void testGatewayMode() throws Exception {
         assertThat(actual).contains("execution.target", "yarn-session");
     }
 
+    @Test
+    void testGatewayModeUrl() throws Exception {
+        String[] args =
+                new String[] {
+                    "gateway",
+                    "-e",
+                    new URL(
+                                    "http",
+                                    SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetAddress(),
+                                    SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetPort(),
+                                    "")
+                            .toString()
+                };
+        String actual = runSqlClient(args, String.join("\n", "SET;", "QUIT;"), false);
+        assertThat(actual).contains("execution.target", "yarn-session");
```

Comment:
Does this verify that the SQL client comes up? Kind of odd to have the yarn-session in there.

Reply:
It does, has to do with how the existing tests are setup:
https://github.com/apache/flink/pull/22556/files#diff-f5e15c60fc56dbc6376b39004c12e41ab6501d130f788ecf421e6f09d597d3daR145-R157
The new one is essentially the same test that just verifies that passing a URL instead of hostname:port works without changing anything else.